jax/numpy/lax_numpy.py

```diff
@@ -2627,6 +2640,14 @@ def argmin(a, axis=None):
   return _argminmax(min, a, axis)
 
 
+@_wraps(onp.nanargmin, lax_description=_NANARG_DOC.format("min"))
+def nanargmin(a, axis=None):
+  nan_mask = isnan(a)
+  a = where(nan_mask, inf, a)
```

Comment:
`np.nanargmin` seems to support integer values, not just float values. Won't this coerce ints to floats? I think you need to add handling for types that aren't subtypes of `np.inexact`, e.g., by just delegating to `argmin`. Similarly for `argmax`.

Reply:
Not really sure if this would cause any bugs, but it's definitely a good idea to avoid unnecessary casts. Thanks!